Stripe/STPPaymentIntent.m

```diff
@@ -0,0 +1,161 @@
+
++ (STPPaymentIntentStatus)statusFromString:(NSString *)string {
+    NSDictionary<NSString *, NSNumber *> *map = @{
+                                                  @"requires_source": @(STPPaymentIntentStatusRequiresSource),
+                                                  @"requires_confirmation": @(STPPaymentIntentStatusRequiresConfirmation),
+                                                  @"requires_source_action": @(STPPaymentIntentStatusRequiresSourceAction),
+                                                  @"processing": @(STPPaymentIntentStatusProcessing),
+                                                  @"succeeded": @(STPPaymentIntentStatusSucceeded),
+                                                  @"requires_capture": @(STPPaymentIntentStatusRequiresCapture),
+                                                  @"canceled": @(STPPaymentIntentStatusCanceled),
+                                                  };
+
+    NSString *key = string.lowercaseString;
+    NSNumber *statusNumber = map[key] ?: @(STPPaymentIntentStatusUnknown);
+    return statusNumber.integerValue;
+}
+
++ (STPPaymentIntentCaptureMethod)captureMethodFromString:(NSString *)string {
+    NSDictionary<NSString *, NSNumber *> *map = @{
+                                                  @"manual": @(STPPaymentIntentCaptureMethodManual),
+                                                  @"automatic": @(STPPaymentIntentCaptureMethodAutomatic),
+                                                  };
+
+    NSString *key = string.lowercaseString;
+    NSNumber *statusNumber = map[key] ?: @(STPPaymentIntentCaptureMethodUnknown);
+    return statusNumber.integerValue;
+}
+
++ (STPPaymentIntentConfirmationMethod)confirmationMethodFromString:(NSString *)string {
+    NSDictionary<NSString *, NSNumber *> *map = @{
+                                                  @"secret": @(STPPaymentIntentConfirmationMethodSecret),
+                                                  @"publishable": @(STPPaymentIntentConfirmationMethodPublishable),
+                                                  };
+
+    NSString *key = string.lowercaseString;
+    NSNumber *statusNumber = map[key] ?: @(STPPaymentIntentConfirmationMethodUnknown);
+    return statusNumber.integerValue;
+}
+
+
+#pragma mark - STPAPIResponseDecodable
+
++ (nullable instancetype)decodedObjectFromAPIResponse:(nullable NSDictionary *)response {
+    NSDictionary *dict = [response stp_dictionaryByRemovingNulls];
+    if (!dict) {
+        return nil;
+    }
+
+    // required fields
+    NSString *stripeId = [dict stp_stringForKey:@"id"];
+    NSString *clientSecret = [dict stp_stringForKey:@"client_secret"];
+    NSNumber *amount = [dict stp_numberForKey:@"amount"];
+    NSString *currency = [dict stp_stringForKey:@"currency"];
+    NSString *rawStatus = [dict stp_stringForKey:@"status"];
+    if (!stripeId || !clientSecret || amount == nil || !currency || !rawStatus || !dict[@"livemode"]) {
```

Comment:
can you do `!amount` here too to be consistent? now I'm questioning the behavior for NSNumber

Reply:
also do we need to fail if `rawStatus` or `livemode` are missing since we have `STPPaymentIntentStatusUnknown` or the default `livemode = YES`?